crates/oci/src/utils.rs

```diff
@@ -0,0 +1,44 @@
+use anyhow::{Context, Result};
+use async_compression::tokio::write::GzipEncoder;
+use std::path::{Path, PathBuf};
+
+/// Create a compressed archive of source, returning its path in working_dir
+pub async fn compressed_archive(source: &Path, working_dir: &Path) -> Result<PathBuf> {
```

Comment:
If it would help simplify things, it would probably be fine if this _wasn't_ `async`. You can drop blocking work into a separate thread pool with tokio's [`spawn_blocking`](https://docs.rs/tokio/1.32.0/tokio/task/fn.spawn_blocking.html).

Reply:
But I was so excited to figure out how to do it all async!  I wouldn't say it has been getting in the way but I do realize it might be safer to go sync/spawn_blocking...

Reply:
I give up on async any time I can as soon as it becomes even mildly inconvenient 😅